spec/system/authentication_and_authorisation_spec.rb

```diff
@@ -1,6 +1,37 @@
 require "spec_helper"
 
 RSpec.describe "Authenication and authorisation" do
+  context "omniauth request phase" do
+    let(:authorize_params) { Rack::Utils.parse_query(URI.parse(page.response_headers["Location"]).query) }
+
+    before do
+      visit "/auth/gds"
+    end
+
+    it "includes pkce code_challenge_method in request for /oauth/authorize" do
+      expect(authorize_params["code_challenge_method"]).to eq("S256")
+    end
+
+    it "includes pkce code_challenge in request for /oauth/authorize" do
+      expect(authorize_params["code_challenge"]).to be_present
+    end
+  end
+
+  context "omniauth callback phase" do
+    it "includes pkce code_verifier in request for /oauth/access_token" do
+      visit "/auth/gds"
+
+      state = Rack::Utils.parse_query(URI.parse(page.response_headers["Location"]).query)["state"]
```

Comment:
Minor: Similar to my previous comment, I think it might be slightly clearer to extract `URI.parse(page.response_headers["Location"])` into a separate local variable and add an assertion that the path is "/oauth/access_token" as we expect.

Reply:
Done :+1: